pkg/spoke/auth/cache_store.go

```diff
@@ -0,0 +1,183 @@
+package auth
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"k8s.io/klog/v2"
+)
+
+func NewExecutorCache() *ExecutorCaches {
+	return &ExecutorCaches{
+		lock:  sync.RWMutex{},
+		items: make(map[string]*DimensionCaches),
+	}
+}
+
+type ExecutorCaches struct {
+	lock sync.RWMutex
+
+	// map key: executor in format of {namespace}/{name}
+	items map[string]*DimensionCaches
+}
+
+type DimensionCaches struct {
+	lock sync.RWMutex
+
+	// map key: hash(Dimension)
+	items map[string]CacheValue
+}
+
+type CacheValue struct {
+	Dimension Dimension
+	Allowed   bool
+}
+
+type Dimension struct {
+	Group         string
+	Version       string
+	Resource      string
+	Namespace     string
+	Name          string
+	ExecuteAction ExecuteAction
+}
+
+// AddNewDimensionCaches adds a dimension caches by the executor if it does not exist, updates it if it exists
+func (c *ExecutorCaches) AddNewDimensionCaches(executor string) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	dimensionCaches := &DimensionCaches{
+		lock:  sync.RWMutex{},
+		items: make(map[string]CacheValue),
+	}
+	c.items[executor] = dimensionCaches
+}
+
+func (c *ExecutorCaches) GetDimensionCaches(executor string) (*DimensionCaches, bool) {
```

Comment:
i am not quite sure about returning point here,  is it safe?

Reply:
changed it to an internal method